src/minimetrics/transport.py

```diff
@@ -0,0 +1,110 @@
+    def encode(self, value: IN) -> OUT:
+        raise NotImplementedError()
+
+
+class RelayStatsdEncoder(MetricEnvelopeEncoder[ExtractedMetric, str]):
+    MULTI_VALUE_SEPARATOR = ":"
+    TAG_SEPARATOR = ","
+
+    def encode(self, value: ExtractedMetric) -> str:
+        metric_name = self._sanitize_value(value["name"])
+        if not metric_name:
+            raise EncodingError("The sanitized metric name is empty")
+        metric_unit = self._get_metric_unit(value.get("unit"))
+        metric_value = self._get_metric_value(value["value"])
+        metric_type = value["type"]
+        metric_tags = self._get_metric_tags(value.get("tags"))
+        metric_timestamp = self._get_metric_timestamp(value["timestamp"])
+
+        return f"{metric_name}{metric_unit}:{metric_value}|{metric_type}{metric_tags}{metric_timestamp}"
+
+    def _sanitize_value(self, value: str) -> str:
+        # Remove all non-alphanumerical chars which are different from _ / .
+        return sanitization_re.sub("", value)
+
+    def _get_metric_unit(self, unit: Optional[MetricUnit]) -> str:
+        if unit is None:
+            return ""
+
+        return f"@{unit}"
+
+    def _get_metric_value(self, value: ExtractedMetricValue) -> str:
+        if isinstance(value, (int, float)):
+            return str(value)
+        elif isinstance(value, List):
+            return self.MULTI_VALUE_SEPARATOR.join([str(v) for v in value])
+        elif isinstance(value, Dict):
+            return self.MULTI_VALUE_SEPARATOR.join([str(v) for v in value.values()])
+
+        raise Exception("The metric value must be either a float or a list of floats")
+
+    def _get_metric_tags(self, tags: Optional[MetricTagsInternal]) -> str:
+        if not tags:
+            return ""
+
+        # We first filter all tags whose sanitized tag key is empty and we also sanitize all tag values. Note that empty
+        # tag values are possible.
+        filtered_tags = (
+            (self._sanitize_value(tag_key), self._sanitize_value(tag_value))
+            for tag_key, tag_value in tags
+            if self._sanitize_value(tag_key)
+        )
+        # We then convert all tag values that are not empty into the string protocol representation.
+        tags_as_string = self.TAG_SEPARATOR.join(
+            [f"{tag_key}:{tag_value}" for tag_key, tag_value in filtered_tags]
+        )
+        return f"|#{tags_as_string}"
+
+    def _get_metric_timestamp(self, timestamp: int) -> str:
+        return f"|T{timestamp}"
+
+
+class MetricEnvelopeTransport(Generic[M]):
+    def __init__(self, encoder: MetricEnvelopeEncoder[M, str]):
+        self._encoder = encoder
+
+    def send(self, metric: M):
+        client = sentry_sdk.Hub.current.client
+        if client is None:
+            return
+
+        try:
+            encoded_metric = self._encoder.encode(metric)
+        except EncodingError:
+            return
+
+        metric_item = Item(payload=encoded_metric, type="statsd")
```

Comment:
_nit_: The implementation is generic, but the base class hard-codes `statsd` as item type. 

As mentioned above, you can keep this simple and have just a class for this all.